distributed/tests/test_worker.py

```diff
@@ -1198,3 +1198,73 @@ def test_avoid_oversubscription(c, s, *workers):
 
     # Some other workers did some work
     assert len([w for w in workers if len(w.outgoing_transfer_log) > 0]) >= 3
+
+
+@gen_cluster(client=True)
+def test_register_init_func(c, s, a, b):
+    #preload function to run
+    def mystartup():
+        import matplotlib.bezier
+
+    def mystartup2():
+        import os
+        os.environ['MY_ENV_VALUE'] = 'WORKER_ENV_VALUE'
+        return "Env set."
+
+    #Check that preload function has been run
+    def test_import():
+        import sys
+        return 'matplotlib.bezier' in sys.modules.keys()
```

Comment:
*Aside*: `'matplotlib.bezier' in sys.modules` should be sufficient. Prob wouldn't bother changing it though unless you're making other changes also.

Reply:
Note that all of the workers are in this process, so these don't currently test what happens when we get a new worker.  I think that this can be resolved by having the function modify the worker itself.